addons/crm/models/crm_lead.py

```diff
@@ -559,6 +560,14 @@ def get_email_to_search(email):
                 duplicate_lead_ids |= return_if_relevant('crm.lead', common_lead_domain + [
                     ('partner_name', 'ilike', lead.partner_name)
                 ])
+            if lead.phone and len(lead.phone) >= MIN_PHONE_LENGTH:
+                duplicate_lead_ids |= return_if_relevant('crm.lead', common_lead_domain + [
+                    '|', ('phone', '=', lead.phone), ('mobile', '=', lead.phone)
+                ])
```

Comment:
I'm wondering, since `phone` is synchronized with the partner, could we move this part in a `elif` after the check on the partner. Like this we could reduce a bit the queries when there is a partner_id set on the lead.

So perhaps something like this:

```python
if lead.partner_id and lead.partner_id.commercial_partner_id:
    ...
elif not lead.partner_id.phone:
    if lead.phone and len(lead.phone) >= MIN_PHONE_LENGTH:
        ...
```

We could also do the same with `mobile` if it's correctly synchronized with the partner but it does not seem to be the case actually. Perhaps there is a reason for this or it's not worth it but it could be cool to check just in case I guess.